### PR TITLE
feat(ngx-layout): provide a service that allows subscribing to one or more media query

### PR DESCRIPTION
**Pull request by WHeirstrate**
# Description

This PR introduces a new service. It allows the user to provide one or more media queries whose changes will be emitted onto a stream. This stream can be subscribed to, but requires an id (the same that was used upon setting the query). If the id of the triggered breakpoint, the id of the triggered query will be emitted onto the stream. Those subscribers whose id matches the id that was pushed onto the stream, will be triggered and will receive the new query value (a boolean).

# Demo

https://github.com/user-attachments/assets/d492f1ec-f169-413d-b5ac-eb7676ec120d

### CODE REVIEW

File: libs/layout/src/lib/services/media-query/mediaquery.service.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+	constructor(private readonly windowService: WindowService) {}
+
+	/**
+	 * Register a list of media queries that need to be tracked by the service.
+	 *
+	 * @param queries - A list of media queries that should be registered with the service.
+	 */
+	public registerMediaQueries(...queries: [id: string, query: string][]): void {
+		this.windowService.runInBrowser(({ browserWindow }) => {
+			for (let [id, query] of queries) {
```
Comment (WHeirstrate) [inline libs/layout/src/lib/services/media-query/mediaquery.service.ts:41]: @IbenTesara remarked:
> I would add an early exit when a listener has already been registered on a certain `id`. Currently, the listener of a certain `id` will be overwritten without removing the previous one. This could lead to memory leaks.
> Example: setting two queries with the same id will result in the latter being destroyed in the onDestroy, while the first one will continue to exist and cause a memory leak.

File: libs/layout/src/lib/services/media-query/mediaquery.service.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+import { WindowService } from '@ngx/utils';
+import { filter, map, Observable, ReplaySubject } from 'rxjs';
+
+/**
+ * A service that can be used to track media queries and their changes. It exposes a method
+ * to register media queries, which takes an array of tuples with the id of the media query
+ * and the query itself. The service will then emit the id of the media query that has
+ * changed when subscribed to the `getMatchingQuery$` method.
+ */
+@Injectable()
```
Comment (WHeirstrate) [inline libs/layout/src/lib/services/media-query/mediaquery.service.ts:11]: @IbenTesara remarked:
> If we'd be supporting multiple matchingQueries, perhaps it would be more sensible to define the queries (and listeners) globally (in the `app.component.ts`), instead of setting them in each component that requires their use.

File: libs/layout/src/lib/services/media-query/mediaquery.service.ts (hunk 1 above)
Comment (WHeirstrate) [reply libs/layout/src/lib/services/media-query/mediaquery.service.ts:41]: Good catch!

File: libs/layout/src/lib/services/media-query/mediaquery.service.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
+import { Injectable, OnDestroy } from '@angular/core';
+import { WindowService } from '@ngx/utils';
```
[the remark this answers is not in the document]
Comment (IbenTesara) [reply libs/layout/src/lib/services/media-query/mediaquery.service.ts:2]: Alternatively, move this to the utils.

File: libs/layout/src/lib/services/media-query/mediaquery.service.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+		});
+	}
+
+	/**
+	 * Pass the id of the query whose changes need to be listened to.
+	 *
+	 * @param id - The id of the media query that should be checked.
+	 * @returns An observable that emits a boolean value whenever the requested media query changes.
+	 */
+	public getMatchingQuery$(id: string): Observable<boolean> {
```
Comment (IbenTesara) [inline libs/layout/src/lib/services/media-query/mediaquery.service.ts:75]: I would throw an error here if no query with the id was registered.

**Review by IbenTesara — CHANGES_REQUESTED** [review]

File: libs/layout/README.md
Comment (IbenTesara) [inline libs/layout/README.md:1]: Maybe just revert the changes here, they're not necessary and just bloat the PR.

File: libs/utils/README.md [hunk 5]
```diff
@@ -14,9 +14,9 @@ npm install @studiohyperdrive/ngx-utils
 
 This package will follow a semver-like format, `major.minor.patch`, in which:
 
-- `major`: Follows the Angular major version
-- `minor`: Introduces new features and (potential) breaking changes
-- `patch`: Introduces bugfixes and minor non-breaking changes
+-   `major`: Follows the Angular major version
```
Comment (IbenTesara) [inline libs/utils/README.md:17]: Is this really necessary? Because now we get an inconsistency between the readmes 🙈 If it's needed it's better to just do them all at once in a separate PR.

**Review by IbenTesara — APPROVED** [review]
PR looks good, just two small comments to take care of before publishing. I'll approve it already though. 👍

File: libs/utils/README.md (hunk 5 above)
Comment (WHeirstrate) [reply libs/utils/README.md:17]: It is not, this was auto changed. I'll revert
